### PR TITLE
Added directory check and exit code 1

### PR DESCRIPTION
**Pull request by LadyDascalie**
Hi there, this contains a quick fix for issue #39.
Love your tool !

### CODE REVIEW

File: ccat.go [hunk 1]
```diff
@@ -68,8 +69,18 @@ func CCat(fname string, p CCatPrinter, w io.Writer) error {
...
+
 		defer file.Close()
-		r = file
+
+		fi, err := file.Stat()
+
+		switch mode := fi.Mode(); {
+		case mode.IsDir():
+			fmt.Println(file.Name() + ": Is a directory")
+			os.Exit(1)
```
Comment (jingweno) [inline ccat.go]: We could return an error here and main.go will take care of exiting the program with the right error code: https://github.com/LadyDascalie/ccat/blob/09c185b4e142906c72de2448a30b40cd703f490d/main.go#L83.

```go
return fmt.Errorf("%s is a directory", file.Name())
```

File: ccat.go [hunk 2]
```diff
@@ -68,8 +69,18 @@ func CCat(fname string, p CCatPrinter, w io.Writer) error {
 		if err != nil {
 			return err
 		}
+
 		defer file.Close()
-		r = file
+
+		fi, err := file.Stat()
+
+		switch mode := fi.Mode(); {
```
Comment (jingweno) [inline ccat.go]: Is `switch` necessary? The input is either a directory or not. So I think `if...else` would be sufficient.

**Comment by jingweno** [comment]
@LadyDascalie Thanks for the contribution. Made a few comments.

**Comment by LadyDascalie** [comment]
Good shout @jingweno !
I reworked the check to be smaller and return an error instead so main can handle it as you suggested! Thanks for the code review, I hope this works for you!

File: ccat.go [hunk 3]
```diff
@@ -68,7 +69,15 @@ func CCat(fname string, p CCatPrinter, w io.Writer) error {
 		if err != nil {
 			return err
 		}
+
 		defer file.Close()
+
+		fi, err := file.Stat()
```
Comment (jingweno) [inline ccat.go]: Check err and return?

**Comment by LadyDascalie** [comment]
Check err and return.

**Comment by yyaabboopp** [comment]
Thanks @LadyDascalie for this fix and @jingweno for accepting it. How long does it take/what is the process to get this pushed upstream to its brew formulae?

Cheers

**Pull request merged**